keras_cv/layers/preprocessing/resizing.py

```diff
@@ -0,0 +1,287 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+import keras_cv.utils
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer,
+)
+
+H_AXIS = -3
+W_AXIS = -2
+
+
+class Resizing(BaseImageAugmentationLayer):
+    """A preprocessing layer which resizes images.
+
+    This layer resizes an image input to a target height and width. The input
+    should be a 4D (batched) or 3D (unbatched) tensor in `"channels_last"`
+    format.  Input pixel values can be of any range (e.g. `[0., 1.)` or `[0,
+    255]`) and of interger or floating point dtype. By default, the layer will
+    output floats.
+
+    This layer can be called on tf.RaggedTensor batches of input images of
+    distinct sizes, and will resize the outputs to dense tensors of uniform
+    size.
+
+    For an overview and full list of preprocessing layers, see the preprocessing
+    [guide](https://www.tensorflow.org/guide/keras/preprocessing_layers).
+
+    Args:
+      height: Integer, the height of the output shape.
+      width: Integer, the width of the output shape.
+      interpolation: String, the interpolation method. Defaults to `"bilinear"`.
+        Supports `"bilinear"`, `"nearest"`, `"bicubic"`, `"area"`, `"lanczos3"`,
+        `"lanczos5"`, `"gaussian"`, `"mitchellcubic"`.
+      crop_to_aspect_ratio: If True, resize the images without aspect
+        ratio distortion. When the original aspect ratio differs from the target
+        aspect ratio, the output image will be cropped so as to return the
+        largest possible window in the image (of size `(height, width)`) that
+        matches the target aspect ratio. By default
+        (`crop_to_aspect_ratio=False`), aspect ratio may not be preserved.
+    """
+
+    def __init__(
+        self,
+        height,
+        width,
+        interpolation="bilinear",
+        crop_to_aspect_ratio=False,
+        pad_to_aspect_ratio=False,
+        bounding_box_format=None,
+        **kwargs,
+    ):
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.crop_to_aspect_ratio = crop_to_aspect_ratio
+        self.pad_to_aspect_ratio = pad_to_aspect_ratio
+        self._interpolation_method = keras_cv.utils.get_interpolation(interpolation)
+        self.bounding_box_format = bounding_box_format
+
+        if pad_to_aspect_ratio and crop_to_aspect_ratio:
+            raise ValueError(
+                "`Resizing()` expects either `crop_to_aspect_ratio` or "
```

Comment:
This comment to me implies that one of the two is required -- maybe instead it should be

"Resizing expects at most one of `crop_to_aspect_ratio` or `pad_to_aspect_ratio` to be True"?